experiments/LES/dycoms.jl

```diff
@@ -0,0 +1,176 @@
+using StaticArrays
+using Test
+using CLIMA
+using CLIMA.Atmos
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using CLIMA.VariableTemplates
+
+const ArrayType = CLIMA.get_array_type()
+
+"""
+  Initial Condition for DYCOMS_RF01 LES
+@article{doi:10.1175/MWR2930.1,
+author = {Stevens, Bjorn and Moeng, Chin-Hoh and Ackerman,
+          Andrew S. and Bretherton, Christopher S. and Chlond,
+          Andreas and de Roode, Stephan and Edwards, James and Golaz,
+          Jean-Christophe and Jiang, Hongli and Khairoutdinov,
+          Marat and Kirkpatrick, Michael P. and Lewellen, David C. and Lock, Adrian and
+          Maeller, Frank and Stevens, David E. and Whelan, Eoin and Zhu, Ping},
+title = {Evaluation of Large-Eddy Simulations via Observations of Nocturnal Marine Stratocumulus},
+journal = {Monthly Weather Review},
+volume = {133},
+number = {6},
+pages = {1443-1462},
+year = {2005},
+doi = {10.1175/MWR2930.1},
+URL = {https://doi.org/10.1175/MWR2930.1},
+eprint = {https://doi.org/10.1175/MWR2930.1}
+}
+"""
+function init_dycoms!(state::Vars, aux::Vars, (x,y,z), t)
+    FT            = eltype(state)
+
+    xvert::FT     = z
+    Rd::FT        = R_d
+
+    # These constants are those used by Stevens et al. (2005)
+    qref::FT      = FT(9.0e-3)
+    q_tot_sfc::FT = qref
+    q_pt_sfc      = PhasePartition(q_tot_sfc)
+    Rm_sfc::FT    = 461.5 #gas_constant_air(q_pt_sfc)
+    T_sfc::FT     = 292.5
+    P_sfc::FT     = 101780 #MSLP
+    ρ_sfc::FT     = P_sfc / Rm_sfc / T_sfc
+
+    # Specify moisture profiles
+    q_liq::FT      = 0
+    q_ice::FT      = 0
+    zb::FT         = 600         # initial cloud bottom
+    zi::FT         = 840         # initial cloud top
+    ziplus::FT     = 875
+    dz_cloud       = zi - zb
+    q_liq_peak::FT = 0.00045     # cloud mixing ratio at z_i
+    if xvert > zb && xvert <= zi
+        q_liq = (xvert - zb)*q_liq_peak/dz_cloud
+    end
+    if xvert <= zi
+        θ_liq = FT(289)
+        q_tot = qref
+    else
+        θ_liq = FT(297.5) + (xvert - zi)^(FT(1/3))
+        q_tot = FT(1.5e-3)
+    end
+
+    # Calculate PhasePartition object for vertical domain extent
+    q_pt  = PhasePartition(q_tot, q_liq, q_ice)
+    Rm    = gas_constant_air(q_pt)
+
+    # Pressure
+    H     = Rm_sfc * T_sfc / grav;
+    p     = P_sfc * exp(-xvert/H);
+    # Density, Temperature
+    T = air_temperature_from_liquid_ice_pottemp_given_pressure(θ_liq, p, q_pt)
```

Comment:
I haven't yet removed this function from export but I prefer it not be used outside of MoistThermo since it has a real chance of being misused (usually in the form of an over-defined system, which is tricky to catch). Furthermore, this function is only used when computing non-equilibrium thermodynamic states (which we don't yet support after the first time-step). While all equilibrium thermodynamic states are subsets of non-equilibrium states, the reverse is not true and this can trigger a very large change in thermodynamic quantities after the first timestep.

If you can, try changing this to
```
ts = LiquidIcePotTempSHumEquil_given_pressure(θ_liq_ice, p, q_tot)
T = air_temperature(ts)
ρ = air_density(ts)
```

This will return an equilibrium thermodynamic state, but you may want to confirm that the other variables you wish you prescribed are still close. @smarras79, is this the case that we discussed, and can you confirm that using `LiquidIcePotTempSHumEquil_given_pressure` worked fine?

Reply:
Correct, the latest driver that I have been using uses what @charleskawczynski suggests above. 
@charleskawczynski please, use this driver as reference: 

[dycoms driver](https://github.com/climate-machine/CLIMA/blob/sm/yt-DYCOMS-MULTI-RATE-NEW-MOIST-THERMO/test/DGmethods/compressible_Navier_Stokes/ex_001_dycoms-imex-FOR-MACIEJ.jl)